File: 백준/Silver/10845. 큐/큐.java
import java.util.*;
public class Main{
    public static void main(String[]args){
        
        Scanner in = new Scanner(System.in);
        int N = in.nextInt();
        Queue <Integer> qu = new LinkedList<>();
        StringBuilder sb = new StringBuilder();
        int temp = 0;
        
        for (int i =0; i<N; i++){
            String m = in.next();
            if(m.equals("push")){
                temp = in.nextInt();
                qu.add(temp);
            }
            else if(m.equals("pop")){
                if(qu.isEmpty()){
                    sb.append("-1").append("\n");
                }
                else{
                    sb.append(qu.poll()).append("\n");
                }
                
            }
            else if(m.equals("size")){
                sb.append(qu.size()).append("\n");
            }
            else if(m.equals("empty")){
                if(qu.isEmpty()){
                    sb.append("1").append("\n");
                }
                else{
                    sb.append("0").append("\n");
                }
            }
            else if(m.equals("front")){
                if(qu.isEmpty()){
                    sb.append("-1").append("\n");
                }
                else{
                    sb.append(qu.peek()).append("\n");
                }
            }
            else if(m.equals("back")){
                if(qu.isEmpty()){
                    sb.append("-1").append("\n");
                }
                else{
                    sb.append(temp).append("\n");
                }
            }

        }
        System.out.print(sb); 
    }
}

File: 백준/Silver/10845. 큐/큐.py
import sys
input = sys.stdin.readline
print = sys.stdout.write
N = int(input().rstrip())
q = []
for i in range(N):
    s = input().rstrip().split()
    if s[0] == "push":
        q.append(s[1])
    elif s[0] == "pop":
        if len(q) == 0:
            print("-1" +"\n")
        else:
            print(q.pop(0)+"\n")
    elif s[0] == "size":
        print(str(len(q))+"\n")
    elif s[0] == "empty":
        if len(q) == 0:
            print("1"+"\n")
        else:
            print("0"+"\n")
    elif s[0] == "front":
        if len(q) == 0:
            print("-1"+"\n")
        else:
            print(q[0] +"\n")
    elif s[0] == "back":
        if len(q) == 0:
            print("-1"+"\n")
        else:
            print(q[-1]+"\n")
